lib/song.rb

```diff
@@ -0,0 +1,58 @@
+class Song
+  extend Concerns::Findable
+
+  attr_accessor :name
+  attr_reader :artist
+  attr_reader :genre
+
+  @@all = []
+
+  def initialize(name, artist = nil, genre = nil)
+    @name = name
```

Comment:
#### Code Quality - Critical
```initialize(name)``` and ```@name = name``` are common for ```Song``` ```Artist``` and ```Genre``` and should therefore be abstracted into a parent class to avoid repetition.